src/Microsoft.AspNet.Mvc.Core/Controllers/DefaultControllerActionArgumentBinder.cs

```diff
@@ -229,18 +221,16 @@ private void ActivateProperties(object controller, Type containerType, Dictionar
             }
         }
 
-        private OperationBindingContext GetOperationBindingContext(
-            ActionContext actionContext,
-            ActionBindingContext bindingContext)
+        private OperationBindingContext GetOperationBindingContext(ControllerContext context)
         {
             return new OperationBindingContext
             {
-                InputFormatters = bindingContext.InputFormatters,
-                ModelBinder = bindingContext.ModelBinder,
-                ValidatorProvider = bindingContext.ValidatorProvider,
+                InputFormatters = context.InputFormatters,
+                ModelBinder = new CompositeModelBinder(context.ModelBinders),
+                ValidatorProvider = new CompositeModelValidatorProvider(context.ValidatorProviders),
                 MetadataProvider = _modelMetadataProvider,
-                HttpContext = actionContext.HttpContext,
-                ValueProvider = bindingContext.ValueProvider,
+                HttpContext = context.HttpContext,
+                ValueProvider = new CompositeValueProvider(context.ValueProviders),
```

Comment:
Not sure if it makes sense, but instead of creating all these composite things does it make more sense to change the data type that's passed around into an `IList<Foo>`? Or does a CompositeFoo ultimately need to be created at some point for each of these anyway.

(This comment applies to several places throughout this change, not just here.)